File: packages/opencommerce-sdk-js/test/fakes.ts
/**
 * A canned-response fake implementing {@link Transport} — this SDK's
 * equivalent of the PHP SDK's own Guzzle `MockHandler` usage. No test in
 * this suite ever touches a real socket.
 */

import type { Transport, TransportResponse } from "../src/transport.ts";

export interface RecordedCall {
  method: string;
  url: string;
  headers: Record<string, string>;
  jsonBody?: Record<string, unknown>;
  timeoutSeconds: number;
}

export class FakeTransport implements Transport {
  readonly calls: RecordedCall[] = [];
  private readonly status: number;
  private readonly body: Record<string, unknown>;

  constructor(status: number, body: Record<string, unknown>) {
    this.status = status;
    this.body = body;
  }

  async request(
    method: string,
    url: string,
    headers: Record<string, string>,
    jsonBody: Record<string, unknown> | undefined,
    timeoutSeconds: number,
  ): Promise<TransportResponse> {
    this.calls.push({ method, url, headers, jsonBody, timeoutSeconds });
    return { status: this.status, body: this.body };
  }
}

File: packages/opencommerce-sdk-python/tests/fakes.py
"""A canned-response fake for :class:`opencommerce_sdk.transport.Transport`
— this SDK's equivalent of the PHP SDK's own Guzzle ``MockHandler`` usage.
No test in this suite ever touches a real socket.
"""

from __future__ import annotations

from typing import Any, Optional


class FakeTransport:
    def __init__(self, status: int, body: dict[str, Any]) -> None:
        self.status = status
        self.body = body
        self.calls: list[dict[str, Any]] = []

    def request(
        self,
        method: str,
        url: str,
        headers: dict[str, str],
        json_body: Optional[dict[str, Any]] = None,
        timeout: int = 30,
    ) -> tuple[int, dict[str, Any]]:
        self.calls.append(
            {"method": method, "url": url, "headers": headers, "json_body": json_body, "timeout": timeout}
        )
        return self.status, self.body
